internal/observability/call.go

```diff
@@ -434,7 +469,18 @@ func (c call) logStreamEvent(err error, success bool, succMsg, errMsg string, ex
 	if success {
 		ce = c.edge.logger.Check(c.levels.success, succMsg)
 	} else {
-		ce = c.edge.logger.Check(c.levels.failure, errMsg)
+		var lvl zapcore.Level
+		if c.levels.useApplicationErrorFailureLevels {
+			lvl = c.levels.failure
+		} else {
+			lvl = c.levels.serverError
+			code := yarpcerrors.FromError(err).Code()
+			if fault := faultFromCode(code); fault == clientFault {
+				lvl = c.levels.clientError
```

Comment:
This code looks pretty much similar to Line206-213, is it possible to move the common code in to a function and use it?